src/isolate/backends/conda.py

```diff
@@ -87,28 +105,47 @@ def create(self) -> Path:
             if env_path.exists():
                 return env_path
 
-            # Since our agent needs Python to be installed (at very least)
-            # we need to make sure that the base environment is created with
-            # the same Python version as the one that is used to run the
-            # isolate agent.
-            dependencies = self._compute_dependencies()
-
-            self.log(f"Creating the environment at '{env_path}'")
-            self.log(f"Installing packages: {', '.join(dependencies)}")
-
-            try:
-                self._run_conda(
-                    "create",
-                    "--yes",
-                    "--prefix",
-                    env_path,
-                    *dependencies,
-                )
-            except subprocess.SubprocessError as exc:
-                raise EnvironmentCreationError("Failure during 'conda create'") from exc
-
-        self.log(f"New environment cached at '{env_path}'")
-        return env_path
+            if self.env_dict:
+                tf = tempfile.NamedTemporaryFile(suffix='.yml')
+                filename = tf.name
+                self.env_dict['dependencies'] = self._compute_dependencies()
+                with open(filename, "w") as f:
+                    yaml.dump(self.env_dict, f)
+                try:
+                    self._run_conda(
+                        "env",
+                        "create",
+                        "-f",
+                        filename,
+                        "--prefix",
+                        env_path
+                    )
```

Comment:
From what I'm seeing in [tempfile](https://github.com/python/cpython/blob/3.11/Lib/tempfile.py#L494-L509), I think you have to `with` the whole tempfile

```suggestion
            if self.env_dict:
                self.env_dict['dependencies'] = self._compute_dependencies()
                with tempfile.NamedTemporaryFile(suffix='.yml') as tf:
                  yaml.dump(self.env_dict, f)
                  try:
                      self._run_conda(
                          "env",
                          "create",
                          "-f",
                          filename,
                          "--prefix",
                          env_path
                      )
```

Reply:
this automatically `close`s the file which in turn deletes it for you

Reply:
Tip: sometimes tempfile buffer a bit on write, so let's also force flush it before conda env create call to ensure the contents are there before we run it

Reply:
For some reason I couldn't get this to work. But doing it outside `with` clause works fine. 🤷

Reply:
Maybe you have to set mode so it is not the default? https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryFile

```
with tempfile.NamedTemporaryFile(mode="w", suffix='.yml') as tf
```